pandas/tests/frame/test_constructors.py

```diff
@@ -2436,12 +2447,41 @@ def test_construction_preserves_tzaware_dtypes(self, tz):
         tm.assert_series_equal(result, expected)
 
     def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
-        # GH#25843
+        # GH#25843, GH#41555, GH#33401
         tz = tz_aware_fixture
-        result = DataFrame({"d": [Timestamp("2019", tz=tz)]}, dtype="datetime64[ns]")
-        expected = DataFrame({"d": [Timestamp("2019")]})
+        ts = Timestamp("2019", tz=tz)
+        ts_naive = Timestamp("2019")
+
+        with tm.assert_produces_warning(FutureWarning):
+            result = DataFrame({0: [ts]}, dtype="datetime64[ns]")
+
+        expected = DataFrame({0: [ts_naive]})
+        tm.assert_frame_equal(result, expected)
+
```

Comment:
maybe worth parameterzing?

Reply:
i think theres a decent chance these can end up in the TestFromScalar class, will need to see how much hassle it is